motionplan/constraint.go

```diff
@@ -166,7 +166,7 @@ func newObstacleConstraint(frame referenceframe.Frame,
 // if reportDistances is false, this check will be done as fast as possible, if true maximum information will be available for debugging.
 func newCollisionConstraint(
 	frame referenceframe.Frame,
-	obstacles map[string]spatial.Geometry,
+	obstacles []spatial.Geometry,
```

Comment:
[q]: unsure we we log collisions that occurred so I cannot validate on my end. Because we switched from map to a list the way we would structure the error log would be slightly different. imo this should be updated if not already so debugging is easier for users.